### ISSUE TITLE
ditto issue with smb drive

### ISSUE THREAD

**Issue by floco** [RhetTbull/osxphotos#85]
Hello,

First let me start by saying osxphotos is just what I was looking for to replace the old ExportPhotosLibrary I was using before. Great work !

I noticed an issue linked to the usage of ditto and remote samba shares. As it can't copy extended attributes it fails without the norsrc option in utils.py. Would you consider adding this as an export option ?
```
subprocess.run(
            ["/usr/bin/ditto", "--norsrc", src, dest], check=True, stderr=subprocess.PIPE
        )
```
also added `import sys` for the below to work
`f"ditto returned error: {e.returncode} {e.stderr.decode(sys.getfilesystemencoding()).rstrip()}"`

Thanks again :-)

PS: I will also look in parallel if there is a way for samba share to accept extended attributes

**Comment by RhetTbull** [comment]
Thanks for alerting me of this issue.  I hadn't considered exporting to non-HFS drives but that's obviously a good use case.  The reason for using ditto was to preserve all metadata and adding --norsrc by default would strip the metadata.  It might be better to to check if destination directory supports metadata and if it does, preserve it, otherwise copy without metadata?  I'll take a look at this and work on a fix.

**Comment by floco** [comment]
oh yes, without norsrc should definitely remain the default, but perhaps there could be a copy_metadata export parameter. Would be true by default and in case it's false then use the --norsrc or an alternative method to copy.
But you are right ideally the destination share should support xattr. In my case, I have a docker samba exposing exposing a couple of ext4 mounted volumed on a NAS that is compatible with both mac and windows. I think the mount should be done with xattr but still need to investigate. I have the same behaviour with home assistant samba addon. So it might be default for several file share solutions.
Thanks again :-)

**Comment by floco** [comment]
Updated the docker image with latest version that implements those guidelines:
https://wiki.samba.org/index.php/Configure_Samba_to_Work_Better_with_Mac_OS_X
ditto works great now :-) in that case what I suggested might not be required. You were right, better fix the root cause.

**Comment by RhetTbull** [comment]
Glad you found a solution! I'll keep the issue open while I work on a solution for filesystems that don't support extended attributes.  Some sort of flag to export, as you suggested, would work.

**Comment by RhetTbull** [comment]
@floco I've added --no-extended-attributes option to export to address this issue

**Comment by floco** [comment]
Perfect 👍 thanks !

**Comment by RhetTbull** [comment]
@floco FYI, I've removed the --no-extended-attribute option in [v0.38.0](https://github.com/RhetTbull/osxphotos/releases/tag/v0.38.0) because I think it's no longer needed.  I implemented a new copy method that takes advantage of copy-on-write on APFS volumes which makes export much faster.   As a result, I don't think the --no-extended-attribute methods is needed any more.  If you run into any issues though, please post a note here or open an issue.

**Comment by floco** [comment]
thanks a lot for keeping me informed 👍
